Services/AspireService.cs

```diff
@@ -1,93 +1,163 @@
+                    continue;
+                }
+
+                var result = labels.TryGetValue("containerapps.io/app-name", out string appName);
+                if (!result)
+                {
+                    _logger.LogDebug($"The Pod {pod.Metadata.Name} does not have an entry for the key app. Skipping to next pod.");
+                    continue;
+                }
+
+                _logger.LogDebug($"App name: {appName}");
+                _logger.LogDebug($"Result: {result}");
+                watchResourcesUpdate.InitialData.Resources.Add(new Resource() { Name = appName, DisplayName = appName, ResourceType = pod.Kind, Uid = pod.Uid(), CreatedAt = Google.Protobuf.WellKnownTypes.Timestamp.FromDateTime(pod.CreationTimestamp().Value), State = pod.Status.Phase, Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = pod.Status.PodIP } }, Properties = { new ResourceProperty() { DisplayName = pod.Kind, Name = pod.Metadata.Name } } });
+            }
+
+            await responseStream.WriteAsync(watchResourcesUpdate);
+        }
+
+        private async Task WatchAndWriteChangesToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Creates the watcher
+            var podsWatchResponse = await _kubernetesClient.CoreV1.ListNamespacedPodWithHttpMessagesAsync(_kubernetesConfig.Namespace, watch: true);
+            var watchResourcesUpdate = new WatchResourcesUpdate() { Changes = new WatchResourcesChanges() };
+            var podWatcher = podsWatchResponse.Watch<V1Pod, V1PodList>(async (type, item) =>
+            {
+                var labels = item.Labels();
+                if (labels.IsNullOrEmpty())
+                {
+                    _logger.LogDebug($"Unable to fetch app name because Pod: {item.Metadata.Name} does not contain labels.Skipping to next pod.");
+                    return;
+                }
+
+                var result = labels.TryGetValue("containerapps.io/app-name", out string appName);
+                if (!result)
+                {
+                    _logger.LogDebug($"The Pod {item.Metadata.Name} does not have an entry for the key app. Skipping to next pod.");
+                    return;
+                }
+
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                if (type.Equals(WatchEventType.Added) || type.Equals(WatchEventType.Modified))
+                {
+                    watchResourcesUpdate.Changes.Value.Add(new WatchResourcesChange() { Upsert = { Name = appName, DisplayName = appName, ResourceType = item.Kind, Uid = item.Uid(), CreatedAt = Google.Protobuf.WellKnownTypes.Timestamp.FromDateTime(item.CreationTimestamp().Value), State = item.Status.Phase, Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = item.Status.PodIP } }, Properties = { new ResourceProperty() { DisplayName = item.Kind, Name = item.Metadata.Name } } } });

```

Comment:
Here for endpoint we're using the item.Status.PodIP, that will not be accessible from outside of the cluster. Instead, can't we get the real endpoints, such as `https://myapp.ashyforest-a8b45123.eastus.azurecontainerapps-test.io`?

Reply:
Just so I'm clear - I'm not finding it in the spec - https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1PodStatus.md. Does it make sense to use host ip?

Reply:
Thanks for clarifying. No specific concern, I wanted to be sure I understood.

Reply:
Sorry, replied to the wrong comment.